src/apps/lwaftr/lwaftr.lua

```diff
@@ -43,6 +44,43 @@ local function compute_binding_table_by_ipv4(binding_table)
    return ret
 end
 
+local function port_to_psid(port, reserved_ports_bit_count, psid_len)
+   local shift = 16 - (reserved_ports_bit_count + psid_len)
+   return band(rshift(port, shift), (2^psid_len)-1)
```

Comment:
I assume you refer to the practice of returning an invalid PSID when the port is in the restricted range, though your comment is short so I have to guess :)  I don't think this is fragile in any significant way (you didn't specify the way either :) and I also think this has an advantage, to wit:

This lookup is on the hot path.  If the port is invalid (in the restricted range), we need to indicate an error somewhere.  This function doesn't have the context to handle the error, but the code that can handle the error can't detect the error on its own.  Fine.  So we have to return a sentinel of some kind.  We could return two values or return optionally nil.  However I think this is not so great for something on the hot path.  Better to return a uint32 and punt the error on down the line -- the error is effectively "packet not found in binding table".  We have to check anyway whether the packet is in the binding table, so we can just give it a PSID that no tunnel will have -- and we know how to do that here.